src/NaNMath.jl

```diff
@@ -22,7 +22,10 @@ sqrt(x::Real) = x < 0.0 ? NaN : Base.sqrt(x)
 # Don't override built-in ^ operator
 pow(x::Float64, y::Float64) = ccall((:pow,libm),  Float64, (Float64,Float64), x, y)
 pow(x::Float32, y::Float32) = ccall((:powf,libm), Float32, (Float32,Float32), x, y)
-pow(x::Number,y::Number) = pow(float(x),float(y))
+function pow(x::Number, y::Number)
+    xf = float(x)
+    return pow(xf, oftype(xf, y))
```

Comment:
If x is `Float32` and y is `Float64`, this will downconvert y to `Float32`. We'd prefer to promote both to `Float64`, right?

Reply:
Ah good point :+1: